src/Tools/AI Test Toolkit/src/AITTestContextImpl.Codeunit.al

```diff
@@ -270,4 +274,59 @@ codeunit 149043 "AIT Test Context Impl."
 
         CurrentTestOutputJson.Add(ElementName, TestInput.GetTestInput(ElementName).ValueAsText());
     end;
+
+    /// <summary>
+    /// Import the Test Input Dataset from the resource file.
+    /// </summary>
+    /// <param name="ResourceFileName">The file name of the resource file which will be used in the description of the dataset.</param>
+    /// <param name="ResourceDataset">The InStream of the resource file.</param>
+    procedure ImportTestInputsOnInstall(ResourceFileName: Text; var ResourceDataset: InStream)
+    var
+        TestInputGroup: Record "Test Input Group";
+        TestInputsManagement: Codeunit "Test Inputs Management";
+        CallerModuleInfo: ModuleInfo;
+    begin
+        // Check if the dataset with the same description exists
+        AITTestSuiteMgt.GetCallerModuleInfo(CallerModuleInfo);
+        TestInputGroup.SetLoadFields(Description, "Imported by AppId");
+        TestInputGroup.SetRange(Description, ResourceFileName);
+
+        if TestInputGroup.FindFirst() then
```

Comment:
This does not seem resillient. You're deleting the first record that match based on the Description field of the record only. What if I have 3 test input groups with the same description?

I think we need to read the file to determine what should be done, or use the filename as a unique ID, which.

Reply:
(Description, "Imported by AppId") is not a unique key

Reply:
@stkillen as discussed, added file name which is not editable. This will be a breaking change.